### DISCUSSION TITLE
revalidatePath and serverless function caching - how to ?

### DISCUSSION THREAD

**Discussion by thibautvdu** [vercel/next.js#66679] [Help]
### Summary

I'm using Next.js 14 with the app router and I'm trying to optimise a specific serverless function usage by caching its answer and invalidating it based on user's data change.

Function : /api/script/[userId]
```
export async function GET(
  request: NextRequest,
  params: {params: {userId: string}},
): Promise<NextResponse> {
  const { userId } = params.params;

  console.log("Revalidated the function for user", { userId });

  return new NextResponse(`hello user ${userId}`, {
    headers: {
      "Vercel-CDN-Cache-Control": "max-age=86400", // 1 day
    },
  });
}
```

And invalidating it with a simple function : /api/invalidate-user-script/[userId]
```
export async function GET(
  request: NextRequest,
  params: {params: {userId: string}},
): Promise<NextResponse> {
  const { userId } = params.params;

  const path = `/api/script/${userId}`;
  revalidatePath(path);

  console.log(`invalidated script cache for user`, { userId, path });

  return new NextResponse();
}
```

However it doesn't work ; curl -I returns a 'HIT' after executing revalidatePath.
Am I missing something ?

**Comment by JeelRajodiya** [comment]
Hello @thibautvdu 
I think your method is correct, are you testing on the deployed version? 

I think in localhost validation and caching does not work properly ([source](https://github.com/vercel/next.js/issues/49778)), can you deploy it and then test it again?

**Comment by thibautvdu** [comment]
Thanks @JeelRajodiya for your reply ; indeed, it fails both on production and preview.

I created a dedicated issue after investigating a bit more, with a reproductible minimum repo : [issue 66680](https://github.com/vercel/next.js/issues/66680)
